test/unittest/tests/block/test_block_ckpt.cpp

```diff
@@ -36,7 +36,9 @@ TEST_CASE("Block helper: __wt_rduppo2", "[block]")
 static void
 test_ckpt_add_blkmod_entry(wt_off_t offset, wt_off_t len, uint64_t expectedBits)
 {
-    std::shared_ptr<SessionWrapper> session = SessionWrapper::buildTestSessionWrapper();
+    auto conn = ConnectionWrapper();
+    auto session = conn.createSession();
```

Comment:
I'm a big fan of `auto`, especially for things like STL iterators, but here, I think `WT_SESSION_IMPL *` would be clearer (as well as doing some type checking).
As we have a 'dumb' pointer, I suggest we should document what is happening here and why we don't need to `delete` it.

Reply:
That's reasonable, although I don't quite follow what you mean by type checking sorry. Can you elaborate on that a bit?

Regarding documentation, I agree but it should probably be part of the `ConnectionWrapper`, since is going to happen anywhere `ConnectionWrapper` is used.

Reply:
Re the type checking, if `auto session` is replaced by `WT_SESSION_IMPL *session` then we get a type error if something goes wrong (eg with the return type of `createSession()` ). I often find specifying the expected type flushes out type errors I've made earlier than when I've used `auto`. Of course, when you need to use a complex type (eg from an STL iterator) `auto` becomes very useful.

And you're right about the best documentation location.